sql/src/main/java/io/crate/exceptions/UnsupportedRelationFeatureException.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.exceptions;
+
+import io.crate.metadata.TableIdent;
+
+import java.util.Collection;
+import java.util.Collections;
+
+public class UnsupportedRelationFeatureException extends ValidationException implements TableScopeException {
```

Comment:
The documentation indicates that the error code is specific to operations on readonly/closed partitions. But the classname here doesn't indicate that at all. Would be nice if we could make this clearer to avoid us from starting to use this exception for other error cases as well. Maybe renaming it to `InaccessibleRelation` or more verbose `OperationOnInaccessibleRelationError`

Reply:
@mfussenegger Would it make more sense to keep this exception specifically for closed/read-only tables, or keep it generally free for use elsewhere and remove that specificity from the documentation? I'd lean towards the latter, since theres nothing concrete in the exception class that deals with only closed/read-only exceptions.

Reply:
@Autophagy I wouldn't make it too generic, otherwise it's harder to judge if it makes sense to use it. I think something along the lines of "relation is not accessible" (excl shard / index state cases) would strike a good balance between to specific and too generic.